native/Avalonia.Native/src/OSX/window.mm

```diff
@@ -1887,18 +1887,7 @@ -(double) getExtendedTitleBarHeight
 
 +(void)closeAll
 {
-    NSArray<NSWindow*>* windows = [NSArray arrayWithArray:[NSApp windows]];
-    auto numWindows = [windows count];
-    
-    for(int i = 0; i < numWindows; i++)
-    {
-        auto window = (AvnWindow*)[windows objectAtIndex:i];
-        
-        if([window parentWindow] == nullptr) // Avalonia will handle the child windows.
-        {
-            [window performClose:nil];
-        }
-    }
+    [[NSApplication sharedApplication] terminate:self];
```

Comment:
As I understand it's possible to cancel termination with applicationShouldTerminate only when NSSupportsSuddenTermination is disabled in plist file.

So, previously it was possible to cancel closeAll (since it was just closing windows), but after this PR it won't be possible unless user disabled NSSupportsSuddenTermination.

Not critical, and I would prefer this behavior for consistency. But it should be kept in mind.

Reply:
Hmm, OK, that's not mentioned in the docs for [applicationShouldTerminate ](https://developer.apple.com/documentation/appkit/nsapplicationdelegate/1428642-applicationshouldterminate?language=objc) or [terminate](https://developer.apple.com/documentation/appkit/nsapplication/1428417-terminate?language=objc) and the docs for [NSSupportsSuddenTermination]([https://developer.apple.com/documentation/bundleresources/information_property_list/nssupportssuddentermination?language=objc) specify "whether the _system_ may terminate".

Are your sure that this applies when calling `terminate` manually and not just the OS during shutdown?

Reply:
@grokys I am not sure actually how it works with programmatic terminating. Should be checked.

Reply:
Given this a quick test and seems to work as expected. Hopefully I didn't mess it up as I'm still learning OSX stuff, but seems to be good.